### DISCUSSION TITLE
DataGrid with Image column from Converter

### DISCUSSION THREAD

**Discussion by vaclav-domin** [dotnet/DataGridExtensions#105] [Q&A] [answered]
Hallo
I have defined AlertColumn

        public static DataGridColumn AlertColumn(string header, int width, string bindingColumn, IValueConverter converter)
        {
            var alertColumn = new DataGridTemplateColumn() { Header = header, Width = width };
            FrameworkElementFactory imageFactory = new FrameworkElementFactory(typeof(Image));
            imageFactory.SetValue(FrameworkElement.WidthProperty, 16.0);
            imageFactory.SetValue(FrameworkElement.HeightProperty, 16.0);
            var bi = new Binding(bindingColumn) { Mode = BindingMode.OneWay, Converter = converter };
            imageFactory.SetValue(Image.SourceProperty, bi);
            var dataTemplate = new DataTemplate() { VisualTree = imageFactory };
            alertColumn.CellTemplate = dataTemplate;
            return alertColumn;
        }

Data type is integer and converter return URI of icon

I hope, that searching will be from alert value (1,2,3). As other columns with converter.

Any idea?

**Comment by tom-englert** [comment, marked as answer]
Template columns need the `SortMemberPath` to be set
